torchrl/collectors/distributed/distributed_collector.py

```diff
@@ -0,0 +1,232 @@
+              num_cpus=None,
+              num_gpus=None,
+              memory=None,
+              object_store_memory=None,
+              resources=None):
+    """
+    Creates an instance of a remote ray class.
+
+    Parameters
+    ----------
+    cls: Python Class
+        Class to be remotely instantiated.
+    num_cpus : int
+        The quantity of CPU cores to reserve for this class.
+    num_gpus  : float
+        The quantity of GPUs to reserve for this class.
+    memory : int
+        The heap memory quota for this class (in bytes).
+    object_store_memory : int
+        The object store memory quota for this class (in bytes).
+    resources: Dict[str, float]
+        The default resources required by the class creation task.
+
+    Returns
+    -------
+    w : func
+        A function to create ray remote class instances.
+    """
+    c = ray.remote(
+        num_cpus=num_cpus,
+        num_gpus=num_gpus,
+        memory=memory,
+        object_store_memory=object_store_memory,
+        resources=resources)(cls)
+    c.is_remote = True
+    return c
+
+
+class DistributedCollector(IterableDataset, ABC):
+    """
+    Class to better handle the operations of ensembles of Collectors.
+
+    Contains common functionality across all collectors sets.
+
+    Parameters
+    ----------
+    collector_class : class
+        A collector class.
+    collector_params : dict
+        Collector class kwargs.
+    remote_config : dict
+        Ray resource specs for the remote collectors.
+    num_collectors: int
+        Total number of collectors in the set (including local collector)
+    """
+
+    def __init__(self,
+                 collector_class,
+                 collector_params,
+                 remote_config=default_remote_config,  # TODO: mutable type as input, not correct
+                 num_collectors=1,
+                 total_frames=1000,  # TODO: is this parameter necessary? it is already specified in each collector.
+                 communication="sync",  # "sync" or "async"
+                 ):
+
+        if not _has_ray:
+            raise RuntimeError(
+                f"ray library not found, unable to create a DistributedCollector. "
+            )
+
+        if communication not in ("sync", "async"):
+            raise ValueError(f"Communication parameter in CollectorSet has to be sync or async.")
+
+        # Monkey patching as_remote and print_remote_collector_info to collector class # TODO: is that ok ?
+        collector_class.as_remote = as_remote
+        collector_class.print_remote_collector_info = print_remote_collector_info
+
+        self.collected_frames = 0
+        self.total_frames = total_frames
+        self.collector_class = collector_class
+        self.collector_params = collector_params
+        self.num_collectors = num_collectors
+        self.remote_config = remote_config
+        self.communication = communication
+
+        # Create a local instance of the collector class.
```

Comment:
are we sure we're not actually collecting any data? Is there a smarter way of doing this?

Reply:
yes, we can just provide policy as an input parameter and assume policy will have the latest version of the weights at any moment.